prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt

```diff
@@ -254,6 +345,146 @@ data class ConfigSchemaElement(
+    }
+
+    override fun merge(overwritingElement: ConfigSchemaElement): ConfigSchemaElement = apply {
+        if (overwritingElement !is ConverterSchemaElement) {
+            throw SchemaException("Overwriting element ${overwritingElement.name} was not a ConverterSchemaElement.")
+        }
         if (overwritingElement.hl7Spec.isNotEmpty()) this.hl7Spec = overwritingElement.hl7Spec
+        super.merge(overwritingElement)
+    }
+}
+
+/**
+ * An element within a Schema.
+ * @property name the name of the element
+ * @property condition a FHIR path condition to evaluate. If false then the element is ignored.
+ * @property required true if the element must have a value
+ * @property schema the name of a child schema
+ * @property schemaRef the reference to the loaded child schema
+ * @property resource a FHIR path that points to a FHIR resource
+ * @property value a list of FHIR paths each pointing to a FHIR primitive value
+ * @property resourceIndex the variable name to store a FHIR collection's index number
+ * @property constants element level constants
+ * @property valueSet a list of key-value pairs used to convert the value property
+ * @property debug log debug information for the element
+ * @property bundleProperty a FHIR path denoting where to store the value
+ */
+@JsonIgnoreProperties
+class FHIRTransformSchemaElement(
+    name: String? = null,
+    condition: String? = null,
+    required: Boolean? = null,
+    schema: String? = null,
+    schemaRef: FHIRTransformSchema? = null,
+    resource: String? = null,
+    value: List<String> = emptyList(),
+    resourceIndex: String? = null,
+    constants: SortedMap<String, String> = sortedMapOf(),
+    valueSet: SortedMap<String, String> = sortedMapOf(),
+    debug: Boolean = false,
+    var bundleProperty: String? = null,
+) : ConfigSchemaElement(
+    name = name,
+    condition = condition,
+    required = required,
+    schema = schema,
+    schemaRef = schemaRef,
+    resource = resource,
+    value = value,
+    resourceIndex = resourceIndex,
+    constants = constants,
+    valueSet = valueSet,
+    debug = debug
+) {
+    override fun validate(validationErrors: MutableList<String>): List<String> {
+        /**
+         * Add an error [msg] to the list of errors.
+         */
+        fun addError(msg: String) {
+            validationErrors.add("[$name]: $msg")
+        }
+
+        when {
+            !schema.isNullOrBlank() && bundleProperty != null ->
+                addError("Schema property cannot be used with the bundleProperty property")
+            schema.isNullOrBlank() && bundleProperty == null ->
+                addError("BundleProperty property is required when not using a schema")
+        }
+
+        return super.validate(validationErrors)
+    }
+
+    override fun merge(overwritingElement: ConfigSchemaElement): ConfigSchemaElement = apply {
+        if (overwritingElement !is FHIRTransformSchemaElement) {
+            throw SchemaException(
```

Comment:
This cannot happen by choice and it was a coding error. Use check here instead.

Reply:
Replaced `if`s with `check`s in latest commit